File: sistema-academico/src/main/java/com/universidad/sistema_academico/entity/Usuario.java
package com.universidad.sistema_academico.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;

@Data
@Entity
@Table(name = "usuarios")
public class Usuario implements UserDetails {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombres;
    private String apellidos;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    private String codigoUsuario;

    @Enumerated(EnumType.STRING)
    private Rol rol;

    private boolean activo = true;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority("ROLE_" + rol.name()));
    }

    @Override
    public String getUsername() {
        return email;
    }

    @Override
    public boolean isEnabled() {
        return activo;
    }
}

File: frontend/src/types/usuario.ts
// Espejo de entity/Usuario.java y entity/Rol.java

export type Rol = 'ESTUDIANTE' | 'DOCENTE' | 'ADMINISTRADOR' | 'DIRECCION'

export interface Usuario {
  id: number
  nombres: string
  apellidos: string
  email: string
  codigoUsuario: string
  rol: Rol
  activo: boolean
}

export const ROL_LABELS: Record<Rol, string> = {
  ESTUDIANTE: 'Estudiante',
  DOCENTE: 'Docente',
  ADMINISTRADOR: 'Administrador',
  DIRECCION: 'Dirección',
}
